src/Build/Evaluation/Evaluator.cs

```diff
@@ -245,6 +250,13 @@ internal class Evaluator<P, I, M, D>
             _sdkResolverService = sdkResolverService;
             _submissionId = submissionId;
             _evaluationProfiler = new EvaluationProfiler(profileEvaluation);
+
+            // Only track the newest project for on disk projects.  There's no way to guarantee the MSBuildNewestProject property
+            // is accurate for in-memory projects
```

Comment:
I don't think this is right. It would be more compatible for the implementation to be defined as "newest imported file that appears on disk", since that's what you'd get out of `$(MSBuildAllProjects)`.